### PR TITLE
FIX symlog scale now shows negative labels.

### LINKED ISSUE #7146: symlog scale no longer shows labels on the negative side
Compare
http://matplotlib.org/devdocs/examples/scales/scales.html (~2.0b4)
and
http://matplotlib.org/examples/scales/scales.html (1.5.3)
no tick labels are drawn on the negative side of the symlog plot in 2.0b4, unlike e.g. 1.5.3 which works fine.
Bisected back to PR #5161, but some commits within the PR don't run so I can't do better.

### LINKED ISSUE #7202: Colorbar with SymmetricalLogLocator : issue when handling only negative values
Tested with matplotlib 1.4.3 and 1.5.3 built from sources - python 2.7.11 - Windows and Linux (64-bit)

To reproduce this issue, choose a set of values X so that  -100<X<-10, then plot a NonUniformImage and attach a colorbar with a SymmetricalLogLocator : no tick or ticklabel will be displayed on the colorbar.

### PR DESCRIPTION
**Pull request by NelleV**
The formatter returned empty strings for negative value.

closes #7146

### CODE REVIEW

**Comment by NelleV** [comment]
For information, I am uncovering major problems in the ticker module.

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -902,14 +902,14 @@ def __call__(self, x, pos=None):
...
         exponent = np.round(fx) if isDecade else np.floor(fx)
-        coeff = np.round(x / b ** exponent)
+        coeff = np.round(abs(x) / b ** exponent)
         if coeff in self.sublabel:
             if not isDecade and self.labelOnlyBase:
-                s = ''
+                return ''
             elif x > 10000:
-                s = '%1.0e' % x
+                s = '%1.0e' % abs(x)
```
Comment (efiring) [inline lib/matplotlib/ticker.py]: Something is still wrong--this doesn't make sense.  It looks like the solution is to remove all of these `abs` calls, and instead set `x = abs(x)` right after saving the sign of  x, a few lines above.

Comment (NelleV) [inline lib/matplotlib/ticker.py]: The LogFormatter is used with the SymmetricalLogLocator so it should (and probably always has) deal with negative value. But the LogLocator doesn't deal with negative values.

Comment (NelleV) [inline lib/matplotlib/ticker.py]: I'll fix the abs stuff. Another solution is not to deal with the sign as it is done in the other formatter (or remove the explicit dealing with sign as it is done right now, I am not sure why)
Honestly, I think this whole part should be refactored and rewritten. I just wanted a quick and easy fix in this PR, as this is a critical bug for 2.0 and don't think we should refactor all this before the release.

**Comment by tacaswell** [comment]
@efiring Can you re-review?

**Comment by NelleV** [comment]
Note: the patch for #7202 relies on this fix.

**Comment by efiring** [comment]
Back ported to v2.x as e2de966.

**Comment by NelleV** [comment]
Thanks @efiring for merging!

**Pull request merged**